src/platform/mt8195/include/platform/lib/memory.h

```diff
@@ -0,0 +1,239 @@
+#define SRAM_TRACE_OFFSET (SRAM_STREAM_OFFSET + SRAM_STREAM_SIZE)
+
+/*4K + 4K +2K + 2K + 4K + 4K = 20KB*/
+#define SOF_MAILBOX_SIZE	(SRAM_INBOX_SIZE + SRAM_OUTBOX_SIZE \
+				+ SRAM_DEBUG_SIZE + SRAM_EXCEPT_SIZE \
+				+ SRAM_STREAM_SIZE + SRAM_TRACE_SIZE)
+
+/* Heap section sizes for module pool */
+#define HEAP_RT_COUNT8		0
+#define HEAP_RT_COUNT16		48
+#define HEAP_RT_COUNT32		48
+#define HEAP_RT_COUNT64		32
+#define HEAP_RT_COUNT128	32
+#define HEAP_RT_COUNT256	32
+#define HEAP_RT_COUNT512	4
+#define HEAP_RT_COUNT1024	4
+#define HEAP_RT_COUNT2048	2
+#define HEAP_RT_COUNT4096	2
+
+
+/* Heap section sizes for system runtime heap */
+#define HEAP_SYS_RT_COUNT64	128
+#define HEAP_SYS_RT_COUNT512	16
+#define HEAP_SYS_RT_COUNT1024	8
+
+/* Heap configuration */
+
+#define HEAP_SYSTEM_BASE	(SRAM1_BASE + SOF_MAILBOX_SIZE)
+#define HEAP_SYSTEM_SIZE	0x6000
+
+#define HEAP_SYSTEM_0_BASE	HEAP_SYSTEM_BASE
+
+#define HEAP_SYS_RUNTIME_BASE	 (HEAP_SYSTEM_BASE + HEAP_SYSTEM_SIZE)
+/*24KB*/
+#define HEAP_SYS_RUNTIME_SIZE \
+	(HEAP_SYS_RT_COUNT64 * 64 + HEAP_SYS_RT_COUNT512 * 512 + \
+	HEAP_SYS_RT_COUNT1024 * 1024)
+
+#define HEAP_RUNTIME_BASE	(HEAP_SYS_RUNTIME_BASE + HEAP_SYS_RUNTIME_SIZE)
+/*48*(16 +32) + 32*(64 128+256) + 4*(512+1024) + 1*2048 = 24832 = 24.25KB*/
+#define HEAP_RUNTIME_SIZE \
+	(HEAP_RT_COUNT8 * 8 + HEAP_RT_COUNT16 * 16 + \
+	HEAP_RT_COUNT32 * 32 + HEAP_RT_COUNT64 * 64 + \
+	HEAP_RT_COUNT128 * 128 + HEAP_RT_COUNT256 * 256 + \
+	HEAP_RT_COUNT512 * 512 + HEAP_RT_COUNT1024 * 1024 + \
+	HEAP_RT_COUNT2048 * 2048 + HEAP_RT_COUNT4096 * 4096)
+
+#define HEAP_BUFFER_BASE	(HEAP_RUNTIME_BASE + HEAP_RUNTIME_SIZE)
+#define HEAP_BUFFER_SIZE \
+	(SRAM1_SIZE - SOF_MAILBOX_SIZE - HEAP_RUNTIME_SIZE - SOF_STACK_TOTAL_SIZE -\
+	HEAP_SYS_RUNTIME_SIZE - HEAP_SYSTEM_SIZE)
+
+#define HEAP_BUFFER_BLOCK_SIZE		0x100
+#define HEAP_BUFFER_COUNT	(HEAP_BUFFER_SIZE / HEAP_BUFFER_BLOCK_SIZE)
+
+#define PLATFORM_HEAP_SYSTEM		1 /* one per core */
+#define PLATFORM_HEAP_SYSTEM_RUNTIME	1 /* one per core */
+#define PLATFORM_HEAP_RUNTIME		1
+#define PLATFORM_HEAP_BUFFER		1
+
+/* Stack configuration */
+#define SOF_STACK_SIZE		0x8000
+#define SOF_STACK_TOTAL_SIZE	SOF_STACK_SIZE /*4KB*/
+#define SOF_STACK_BASE		(SRAM1_BASE + SRAM1_SIZE)
+#define SOF_STACK_END		(SOF_STACK_BASE - SOF_STACK_TOTAL_SIZE)
+
+/* Vector and literal sizes - not in core-isa.h */
+#define SOF_MEM_VECT_LIT_SIZE		0x4
+#define SOF_MEM_VECT_TEXT_SIZE		0x1c
+#define SOF_MEM_VECT_SIZE		(SOF_MEM_VECT_TEXT_SIZE + SOF_MEM_VECT_LIT_SIZE)
+
+#define SOF_MEM_RESET_TEXT_SIZE		0x2e0
+#define SOF_MEM_RESET_LIT_SIZE		0x120
+#define SOF_MEM_VECBASE_LIT_SIZE	0x178
+
+#define SOF_MEM_RO_SIZE			0x8
+
+#define HEAP_BUF_ALIGNMENT		DCACHE_LINE_SIZE
+
+/** \brief EDF task's default stack size in bytes. */
+#define PLATFORM_TASK_DEFAULT_STACK_SIZE	3072
+
+#if !defined(__ASSEMBLER__) && !defined(LINKER)
+
+struct sof;
+
+/**
+ * \brief Data shared between different cores.
+ * Does nothing, since IMX doesn't support SMP.
```

Comment:
Update comment - remove IMX.
Please check other places too (like, line 220).

Reply:
Thanks,  I will correct this and update.